Userland/Utilities/unzip.cpp

```diff
@@ -140,12 +160,26 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
         }
 
         if (keep_file) {
+            if (zip_member.is_directory) {
+                zip_directories.append(zip_member);
+            }
             if (!unpack_zip_member(zip_member, quiet))
                 return IterationDecision::Break;
         }
 
         return IterationDecision::Continue;
     }));
 
+    if (!success) {
+        return 1;
+    }
+
+    for (auto& directory : zip_directories) {
+        if (adjust_modification_time(directory).is_error()) {
+            warnln("Failed setting modification time for directory {}", directory.name);
+            return 1;
+        }
+    }
```

Comment:
Don't we have to iterate the directories in the reverse order here? I'd assume that the top-most directory would be listed first in the file, so we'd have to set its modification time last.

Reply:
I don't know if I get your point. It shouldn't matter in which order we go through the saved `ZipMember` entries and set the modication times for a file based on the path stored in `zip_member.name`.

Reply:
Won't setting the modification time of a child directory also update the modification date of the parent directory (as in, "update to the current time", not "update to whatever value was given")? If that isn't the case then don't mind me.

Reply:
No this should only happen when subdirs / files are added / removed.

Made one more small change to only add a directory to the vector if unpacking was actually successful.
Should all be good now :)